modules/tinymce/src/core/test/ts/browser/UndoManagerTest.ts

```diff
@@ -682,4 +683,26 @@ describe('browser.tinymce.core.UndoManagerTest', () => {
     });
   });
 
+  it('TINY-9222: Scroll to the cursor after undo and redo', () => {
+    const editor = hook.editor();
+    editor.undoManager.clear();
+    editor.focus();
+
+    const HEIGHT = 5000;
+    editor.resetContent(`<p class="first">top paragraph</p><p style="height: ${HEIGHT}px"></p><p class="last">last paragraph</p>`);
```

Comment:
`editor.resetContent` clears undo history internally

Reply:
@vpyshnenko Not sure what should I update here, that's kind of the point. Can I ask you to elaborate, please?

Reply:
I wonder if this means you don't need the `editor.undoManager.clear()` above since `resetContent()` already does that?

Reply:
On that note, the `editor.focus()` should also be unnecessary the `TinyHooks` logic is already configured to focus the editor on init.

Reply:
Oh, I see. Replaced `resetContent` with `setContent` to explicitly underline the intention of clearing the `undoManager` [here](https://github.com/tinymce/tinymce/pull/8205/commits/6ce4ec0751aacae05b4a1819b09be4689d7ad628).

Reply:
@yacodes that's not the right thing to do... you should remove the `clear` call instead as now you have a case where there is no initial undo state which definitely isn't what we want here.

Reply:
@lnewson I do not understand. That's exactly what I want:
1. Start with an empty undo state;
2. Add content to the editor to add undo state;
3. Scroll up;
4. Undo;
5. Check if the scroll is down.

Without `editor.undoManager.clear()` the `undo` state is not empty I believe. Other tests are clearing the undo state as well.

Reply:
@yacodes that's not what you want... take a look at a clean editor, it will always have one value in the editor state (the initial state). If it has zero then it's in an invalid state and `undo` will not be allowed/work properly until there are at least 2 data objects in the undo manager. For example, in fiddle run clear and then click the bold button and you'll see the undo button is still disabled. So using `resetContent` is what you want here as it will put the editor back into a "clean" state where you cannot undo or redo without a change.

So while the steps are what I'd expect, the first step isn't right/what's happening here and to correct that using `resetContent` is the right API to use. Also remember to remove `editor.focus()` as I said above unless there's a good reason for it.

Reply:
@lnewson I see the difference now, thanks for the clarification. Updated [here](https://github.com/tinymce/tinymce/pull/8205/commits/2e4b85db39f93e3d2a815f21a8bee3cb80e687cc)